EMS/EMS/DataBase.h

```diff
@@ -23,6 +23,11 @@ class DataBase : public IDataBase {
     DataBase() { engine_ = new SearchEngine(); };
     ~DataBase() = default;
 
+    void setSearchEngine(ISearchEngine* engine) { 
+        delete engine_; 
+        engine_ = (SearchEngine*)engine;
```

Comment:
cpp 캐스팅으로 바꾸는 것을 제안드립니다.
```suggestion
        engine_ = static_cast<SearchEngine*>(engine);
```